File: dialog/src/main/java/com/dialog/OnDialogClickListener.java
package com.dialog;

import android.content.DialogInterface;

/**
 * This interface is used as a call back for the dialogs
 */
public interface OnDialogClickListener {

    enum Button {
        POSITIVE,
        NEGATIVE,
        NEUTRAL
    }

    void onClick(int tag,DialogInterface dialog, int pos, Button button);

}

File: dialog/src/main/java/com/dialog/OnDialogClickListener.kt
package com.dialog

import android.content.DialogInterface

/**
 * This interface is used as a call back for the dialogs
 */
@FunctionalInterface
interface OnDialogClickListener {

    enum class Button {
        POSITIVE,
        NEGATIVE,
        NEUTRAL
    }

    fun onClick(tag: Int, dialog: DialogInterface, pos: Int, button: Button)

}
